lib/turn.rb

```diff
@@ -0,0 +1,84 @@
+require "./lib/card.rb"
+require "./lib/deck.rb"
+require "./lib/player.rb"
+
+class Turn 
+  attr_reader :player1,
+              :player2,
+              :spoils_of_war 
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+
+  def type 
+    if player1.deck.cards.length >= 3 && player2.deck.cards.length >= 3
+      if player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+        :mutually_assured_destruction
+      elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)
+        :war 
+      else 
+        :basic 
+      end 
+    else 
+      if player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0)
+        :war 
+      else 
+        :basic 
+      end 
+    end
+  end
+
+  def winner 
```

Comment:
Similar to the `turn` method, I'm seeing a lot of nested logic here. Try playing around with helper methods to see if you can refactor this to be a little more digestible and easier to test.